documentation/VSTestIntegration.md

```diff
@@ -0,0 +1,92 @@
+# Coverlet Integration with VSTest
+
+## Motivation
+The cross platform solution for Code Coverage generation for .NET Core projects(in a consistent manner).
+The asks for Code Coverage support for .NET Core on Linux is the most commented issue on vstest repo:
+https://github.com/Microsoft/vstest/issues/981#issuecomment-320311552
+
+## Summary
+What would integrating Coverlet with Microsoft Test Platform mean:
+
+1. Coverlet based coverage solution is available as a data collector that does the instrumentation of the necessary modules before test execution and appropriate restore after.
+2. Authoring - When new test projects are created (dotnet mstest), references to the data collector package are added by default. This reduces adoption friction for customers.
+3. Test execution - Today Coverlet is invoked as a msbuild target and as such only works with dotnet test (and requires a csproj). With the data collector becoming available, coverage can also be collected when tests are run on built binaries (dotnet vstest)
+
+## Proposed Solution
+
+### Scenarios to support
+The following table summarizes the support that needs to be added for a seamless code coverage collection for .NET Core on both Linux and Windows platforms:
+
+| Entry point | How will code coverage be enabled? | Syntax                                                               |
+|-------------|------------------------------------|----------------------------------------------------------------------|
+|dotnet test CLI              | Through a switch to condition data collection | `dotnet test --collect:"XPlat Code Coverage"`   |
+|dotnet vstest CLI            | Through a switch to condition data collection | `dotnet vstest --collect:"XPlat Code Coverage"` |
+
+### Coverlet Options Supported with VSTest
+
+#### Default
+| Option | Summary |
+|-------------|------------------------------------|
+|Format              | Results format in which coverage output is generated. Default format is cobertura.| 
```

Comment:
Should we provide default `CoverletOutput`?
With msbuild we are using `MSBuildProjectDirectory` https://github.com/tonerdo/coverlet/blob/0956d1e8da6b7265b0b5e4f93af3499c4be99cc8/src/coverlet.msbuild.tasks/coverlet.msbuild.props#L14 but there is some discussion in place https://github.com/tonerdo/coverlet/issues/152

Reply:
We will only be getting the coverage output from coverlet.core APIs and the responsibility of sending the file will be with the datacollector. 
By default, currently the files are put into the TestResults directory, Where this TestResults directory will be created is configurable via runsettings.